tox.ini

```diff
@@ -2,7 +2,7 @@
 skipsdist = True
 usedevelop = True
 
-envlist = py{38,39,310,311}-dj{32,41,42,42stable,main}-{sqlite,postgres,mysql,mssql}-{elasticsearch8,elasticsearch7,elasticsearch6,elasticsearch5,noelasticsearch}-{customuser,emailuser}-{tz,notz},
+envlist = py{38,39,310,311,312}-dj{32,41,42,42stable,main}-{sqlite,postgres,mysql,mssql}-{elasticsearch8,elasticsearch7,elasticsearch6,elasticsearch5,noelasticsearch}-{customuser,emailuser}-{tz,notz},
```

Comment:
Not specific to this PR, just wanted to note that we don't use tox on the CI and I personally don't use it. Might be worth looking into reusing this on the CI in the future.

Reply:
Yeah, was wondering about that while updating this - I suspect the tox config is a hangover from some earlier CI setup like Travis. Does it provide any value to us, or should we just ditch it?

Reply:
We could make use of it to expand our matrix without necessarily spinning up more runner instances. For example, we could make our GitHub Actions matrix just configures the Python + DB combo, and let tox install all Django versions and test against them. That will make the CI run longer for probably little benefit though.

There's also a mention of it on the docs, not sure if anyone actually uses it.
https://docs.wagtail.org/en/stable/contributing/developing.html#running-only-some-of-the-tests